### PR TITLE
Automated tests for AlloyDB on-demand backup 

### PR DESCRIPTION
**Pull request by GauravJain21**
…sses

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
Added automated tests for AlloyDB on-demand backup

```

### CODE REVIEW

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like the test is failing in CI, are you seeing it pass locally?

**Comment by GauravJain21** [comment]
Hey @slevenick 
All the tests are passing. Can you please review, approve, and merge the PR?

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go [hunk 1]
```diff
@@ -157,3 +157,78 @@ resource "google_service_networking_connection" "vpc_connection" {
...
+  backup_id    = "tf-test-alloydb-backup%{random_suffix}"
+  location = "us-central1"
+  cluster_name = google_alloydb_cluster.default.name
+  depends_on = [google_alloydb_instance.default]
+}
+
+resource "google_alloydb_cluster" "default" {
+  location = "us-central1"
+  cluster_id = "tf-test-alloydb-cluster%{random_suffix}"
+  network    = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go:193]: These can be replaced with `network = google_compute_network.default.id` I believe. Does that work?

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go [hunk 2]
```diff
@@ -157,3 +157,78 @@ resource "google_service_networking_connection" "vpc_connection" {
...
+
+  depends_on = [google_service_networking_connection.vpc_connection]
+}
+
+resource "google_compute_global_address" "private_ip_alloc" {
+  name          =  "tf-test-alloydb-cluster%{random_suffix}"
+  address_type  = "INTERNAL"
+  purpose       = "VPC_PEERING"
+  prefix_length = 16
+  network       = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go:215]: Again here

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go [hunk 3]
```diff
@@ -157,3 +157,78 @@ resource "google_service_networking_connection" "vpc_connection" {
...
+		id,
+		network,
+		project,
+		self_link
+	]
+  }
+}
+
+resource "google_service_networking_connection" "vpc_connection" {
+  network                 = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go:229]: And here

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go (hunk 1 above)
Comment (GauravJain21) [reply mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go]: done

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go (hunk 2 above)
Comment (GauravJain21) [reply mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go]: done

File: mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go (hunk 3 above)
Comment (GauravJain21) [reply mmv1/third_party/terraform/tests/resource_alloydb_backup_test.go]: done

**Review by slevenick — APPROVED** [review]

**Pull request merged**